src/containers/AssignmentTexterContact.jsx

```diff
@@ -695,55 +693,55 @@ class AssignmentTexterContact extends React.Component {
     )
   }
 
-  renderCorrectSendButton(){
-    const { campaign, assignment, texter } = this.props
+  renderCorrectSendButton() {
+    const { campaign } = this.props
     const { contact } = this.props.data
-    let button = null
-    if(contact.messageStatus === 'needsResponse' || contact.messageStatus === 'messaged') {
-      button =
+    if (contact.messageStatus === 'needsResponse' || contact.messageStatus === 'messaged') {
+      return (
         <SendButtonArrow
           threeClickEnabled={campaign.organization.threeClickEnabled}
           onFinalTouchTap={this.handleClickSendMessageButton}
           disabled={this.state.disabled}
         />
-      return button
-    } else if (contact.messageStatus === 'needsMessage') {
-      return button
+      )
     }
-    return button
+    return null
   }
 
   renderBottomFixedSection() {
-    const { assignment, campaign } = this.props
-    const { contact } = this.props.data
+    const { optOutDialogOpen } = this.state
+
+    const message = (optOutDialogOpen) ? '' : (
+      <div className={css(styles.messageField)}>
+        <GSForm
+          ref='form'
+          schema={this.messageSchema}
+          value={{ messageText: this.state.messageText }}
+          onSubmit={this.handleMessageFormSubmit}
+          onChange={this.handleMessageFormChange}
+        >
+          <Form.Field
+            name='messageText'
+            label='Your message'
+            multiLine
+            fullWidth
+            onKeyUp={(event) => {
+              if (event.keyCode === 13) {
+                // pressing the Enter key submits
+                this.handleClickSendMessageButton()
+              }
+            }}
+          />
+          {this.renderCorrectSendButton()}
+        </GSForm>
+      </div>
+    )
 
     return (
       <div>
         {this.renderSurveySection()}
         <div>
-          <div className={css(styles.messageField)}>
-            <GSForm
-              ref='form'
-              schema={this.messageSchema}
-              value={{ messageText: this.state.messageText }}
-              onSubmit={this.handleMessageFormSubmit}
-              onChange={this.handleMessageFormChange}
-            >
-              <Form.Field
-                name='messageText'
-                label='Your message'
-                multiLine
-                fullWidth
-                onKeyUp={(event) => {
-                  if (event.keyCode === 13) {
-                    // pressing the Enter key submits
-                    this.handleClickSendMessageButton()
-                  }
-                }}
-              />
-              {this.renderCorrectSendButton()}
-            </GSForm>
-          </div>
+          {message}
           {this.renderActionToolbar()}
```

Comment:
I think rather than bring in the state to renderActionToolbar, I would vote that the logic here be something like:
{(this.state.optOutDialogOpen ? this.renderActionToolbar() : '')} (or however that's done in React/jsx-ese
My thinking is that the action toolbar as a component (I understand it's not (yet) one), shouldn't "know" about the optout dialog -- it should just be called when it should render or not.  I think it'll also be a little clearer in the code with the control flow that way, since if someone is reading the 'top' render view, then know how it works, rather than having to go down to the sub-(pseudoish)-components